view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js

```diff
@@ -28,6 +28,41 @@ define(
                 let paypalConfiguration = Object.assign(baseComponentConfiguration, paymentMethodsExtraInfo[paymentMethod.type].configuration);
                 paypalConfiguration.showPayButton = true;
 
+                /*
+                 * Add onInit and onClick functions to the configuration object in order to maange the PayPal Buttons
+                 * state (enabled / disabled) depending on the checkout agreements checkbox state.
+                 *
+                 * Without these functions, if the form is not valid (checkbox unchecked) and the user clicks on the
+                 * PayPal button, the popup will be opened with an infinite loading spinner, and even if the customer
+                 * closes the popup, checks the checkbox and clicks again on the PayPal button, the popup will not work
+                 * anymore, until the page is reloaded.
+                 */
+                let checkoutAgreementsInput = document.querySelector('.checkout-agreement input');
```

Comment:
Race condition occurs if the payment method page is refreshed. See my comment [here](https://github.com/Adyen/adyen-magento2/pull/2659#issuecomment-2337670313).